src/Protocols/Internal.hs

```diff
@@ -619,10 +620,10 @@ instance (C.NFDataX a, C.ShowX a, Show a, C.KnownDomain dom) => Drivable (CSigna
   driveC _conf [] = error "CSignal.driveC: Can't drive with empty list"
   driveC SimulationConfig{resetCycles} fwd0@(f:_) =
     let fwd1 = C.fromList_lazy (replicate resetCycles f <> fwd0 <> repeat f) in
-    Circuit ( \_ -> ((), CSignal fwd1) )
+    Circuit ( \_ -> ((), fwd1) )
 
   sampleC SimulationConfig{resetCycles, ignoreReset} (Circuit f) =
-    let sampled = CE.sample_lazy ((\(CSignal s) -> s) (snd (f ((), def)))) in
+    let sampled = CE.sample_lazy (id (snd (f ((), def)))) in
```

Comment:
Is this `id` needed? From what I can see it could be simplified to this, no?
```suggestion
    let sampled = CE.sample_lazy (snd (f ((), def))) in
```